### ISSUE TITLE
VTK Output in parallel has some problems.

### ISSUE THREAD

**Issue by aeslaughter** [libMesh/libmesh#882]
In MOOSE run the simple diffusion test with two processors. The following *.vtu files are produced, the second has all zeros:

simple_diffusion_out_001_0.vtu
```
<?xml version="1.0"?>
<VTKFile type="UnstructuredGrid" version="0.1" byte_order="LittleEndian" header_type="UInt32" compressor="vtkZLibDataCompressor">
  <UnstructuredGrid>
    <Piece NumberOfPoints="67" NumberOfCells="50">
      <PointData>
        <DataArray type="Float64" Name="u" format="ascii" RangeMin="0" RangeMax="1">
          0 0.1 0.1 0 0.2 0.2
          0.3 0.3 0.4 0.4 0.5 0.5
          0.6 0.6 0.7 0.7 0.8 0.8
          0.9 0.9 1 1 0.099999999999 0
          0.2 0.3 0.4 0.5 0.6 0.7
          0.8 0.9 1 0.1 0 0.2
          0.3 0.4 0.5 0.6 0.7 0.8
          0.9 1 0.1 0 0.2 0.3
          0.4 0.5 0.6 0.7 0.8 0.9
          1 0.1 0 0.2 0.3 0.4
          0.5 0.6 0.7 0.8 0.1 0
          0.2
        </DataArray>
      </PointData>
      <CellData>
```

simple_diffusion_out_001_1.vtu
```
<?xml version="1.0"?>
<VTKFile type="UnstructuredGrid" version="0.1" byte_order="LittleEndian" header_type="UInt32" compressor="vtkZLibDataCompressor">
  <UnstructuredGrid>
    <Piece NumberOfPoints="67" NumberOfCells="50">
      <PointData>
        <DataArray type="Float64" Name="u" format="ascii" RangeMin="0" RangeMax="0">
          0 0 0 0 0 0
          0 0 0 0 0 0
          0 0 0 0 0 0
          0 0 0 0 0 0
          0 0 0 0 0 0
          0 0 0 0 0 0
          0 0 0 0 0 0
          0 0 0 0 0 0
          0 0 0 0 0 0
          0 0 0 0 0 0
          0 0 0 0 0 0
          0
        </DataArray>
      </PointData>
      <CellData>
```

**Comment by jwpeterson** [comment]
This is not related to your reproduction test, but it's also not good news for people using AMR with this Output class:
```
/*
 * FIXME: This is known to write nonsense on AMR meshes
 * and it strips the imaginary parts of complex Numbers
 */
void VTKIO::system_vectors_to_vtk(const EquationSystems & es,
                                  vtkUnstructuredGrid *& grid)
```

**Comment by jwpeterson** [comment]
Fixed by #886.
